src/tensors/cpu/prod.cpp

```diff
@@ -134,6 +134,54 @@ void ProdBatched(marian::Tensor C,
   auto strideC = n * m;
 
   auto batchC = std::max(batchA, batchB);
+#if MKL_FOUND
+  CBLAS_TRANSPOSE transA_forarr = CblasNoTrans;
+  CBLAS_TRANSPOSE transB_forarr = CblasNoTrans;
+
+  if(transA)
+    transA_forarr = CblasTrans;
+
+  if(transB)
+    transB_forarr = CblasTrans;
+
+  static const constexpr size_t group_count = 1;
```

Comment:
Please comment what this is and does.
Also please give a general comment how all these different vectors interact and what the batched gemm is supposed to do.